dali/kernels/imgproc/convolution/convolution_gpu.h

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_CONVOLUTION_CONVOLUTION_GPU_H_
+#define DALI_KERNELS_IMGPROC_CONVOLUTION_CONVOLUTION_GPU_H_
+
+#include "dali/core/boundary.h"
+#include "dali/core/convert.h"
+#include "dali/core/format.h"
+#include "dali/core/tensor_view.h"
+#include "dali/kernels/common/utils.h"
+#include "dali/kernels/imgproc/convolution/cutlass/device/gemm.h"
+#include "dali/kernels/kernel.h"
+#include "dali/pipeline/util/operator_impl_utils.h"
+
+namespace dali {
+namespace kernels {
+
+/**
+ * @brief Apply convolution with 1-channel `window` in specified axis.
+ *
+ * Operation is done by using GEMM and generating the matrix from convolution window on the fly
+ */
+template <typename Out, typename In, typename W, int ndim, int axis, bool has_channels = true>
+struct ConvolutionGpu {
+  KernelRequirements Setup(KernelContext& ctx, const TensorListShape<ndim>& in_shape,
+                           const TensorListShape<1>& window_size) {
```

Comment:
Anchor points are not configurable.

Reply:
Yes. I can add this after we add the operator. For what do we need that?

Reply:
Audio?
Being a replacement for OpenCV?
It's just not generic enough. I don't like this being done "for now" and never fixed.